tx/textual/internal/testdata/repeated.json

```
@@ -0,0 +1,165 @@
+            {"text": "String messages: 1 String", "indent": 1},
+            {"text": "String messages (1/1): /OmniFlix.onft.v1beta1.MsgTransferONFT", "indent": 2},
+            {"text": "End of String messages", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "string_messages": [
+                "/OmniFlix.onft.v1beta1.MsgTransferONFT",
+                "/OmniFlix.onft.v1beta1.MsgBurnONFT",
+                "/OmniFlix.marketplace.v1beta1.MsgListNFT"
+            ]
+        },
+        "parses": true,
+        "screens": [
+            {"text": "Baz object", "indent": 0},
+            {"text": "String messages: 3 String", "indent": 1},
+            {"text": "String messages (1/3): /OmniFlix.onft.v1beta1.MsgTransferONFT", "indent": 2},
+            {"text": "String messages (2/3): /OmniFlix.onft.v1beta1.MsgBurnONFT", "indent": 2},
+            {"text": "String messages (3/3): /OmniFlix.marketplace.v1beta1.MsgListNFT", "indent": 2},
+            {"text": "End of String messages", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "vote" : {
+                "proposal_id": 4,
+                "voter": "cosmos1abc...def",
+                "options": [
+                    {
+                        "option": "Yes",
+                        "weight": "0.7"
+                    },
+                    {
+                        "option": "No",
+                        "weight": "0.3"
+                    }
+                ]
+            }
+        },
+        "parses": false,
+        "screens": [
+            {"text": "Baz object", "indent": 0},
+            {"text": "Vote: Ballot object", "indent": 1},
+            {"text": "Proposal id: 4", "indent": 2},
+            {"text": "Voter: cosmos1abc...def", "indent": 2},
+            {"text": "Options: 2 WeightedBallotOption", "indent": 2},
+            {"text": "Options (1/2): WeightedBallotOption object", "indent": 3},
+            {"text": "Option: Yes", "indent": 4},
+            {"text": "Weight: 0.7", "indent": 4},
+            {"text": "Options (2/2): WeightedBallotOption object", "indent": 3},
+            {"text": "Option: No", "indent": 4},
+            {"text": "Weight: 0.3", "indent": 4},
+            {"text": "End of Options", "indent": 2}
+        ]
+    },
+    {
+        "proto": {
+            "price": [
+                { "amount": "1", "denom": "ucosm" },
+                { "amount": "3", "denom": "ustake" }
+            ]
+        },
+        "metadata": {
+            "ucosm": {"display": "COSM", "denom_units": [{"denom": "COSM", "exponent": 6}, {"denom": "ucosm", "exponent": 0}]},
+            "ustake": {"display": "STAKE", "denom_units": [{"denom": "STAKE", "exponent": 6}, {"denom": "ustake", "exponent": 0}]}
+          },
+        "parses": false,
+        "screens": [
+            {"text": "Baz object", "indent": 0},
+            {"text": "Price: 0.000001 COSM, 0.000003 STAKE", "indent": 1}
```

In my opinion, we don't need to test coins with Metadata (they are anyways optional). We can make the JSON less cluttered with simply:

```suggestion
        "parses": false,
        "screens": [
            {"text": "Baz object", "indent": 0},
            {"text": "Price: 1 ucosm, 3 ustake", "indent": 1}
```

But also, since it's here already, totally fine to keep it as is, your call.

---

i wanted to avoid metadata as well but i wanted to make sure that the rendering was properly wired up.

should be easy enough to omit.